snapshots/manager.go

```diff
@@ -371,3 +405,45 @@ func IsFormatSupported(snapshotter types.ExtensionSnapshotter, format uint32) bo
 	}
 	return false
 }
+
+// SnapshotIfApplicable takes a snapshot of the current state if we are on a snapshot height. 
+// It also prunes any old snapshots. The snapshotting and pruning happen in separate goroutines.
+func (m *Manager) SnapshotIfApplicable(height int64) {
+	if m == nil {
+		return
+	}
+	if !m.shouldTakeSnapshot(height) {
+		m.logger.Debug("snapshot is skipped", "height", height)
+		return
+	}
+	go m.snapshot(height)
+}
+
+// shouldTakeSnapshot returns true is snapshot should be taken at height.
+func (m *Manager) shouldTakeSnapshot(height int64) bool {
+	return m.opts.Interval > 0 && uint64(height)%m.opts.Interval == 0
+}
+
+func (m *Manager) snapshot(height int64) {
+	m.logger.Info("creating state snapshot", "height", height)
+
+	snapshot, err := m.Create(uint64(height))
```

Comment:
Please defensively guard against negative values before creating snapshots otherwise I can poison this code with math.MinInt64 which will turn into a massive positive value of 9223372036854775808 and make this code hang and run out of resources https://go.dev/play/p/lLbl3oQcXia

Reply:
Thanks for the defensive check on height, now LGTM!